go/flagutil/optional.go

```diff
@@ -38,100 +39,57 @@ type OptionalFlag interface {
 }
 
 var (
-	_ OptionalFlag = (*OptionalFloat64)(nil)
-	_ OptionalFlag = (*OptionalString)(nil)
+	_ OptionalFlag = (*OptionalType)(nil)
 )
 
-// OptionalFloat64 implements OptionalFlag for float64 values.
-type OptionalFloat64 struct {
-	val float64
+type OptionalType struct {
+	val any
 	set bool
 }
```

Comment:
```
func (t OptionalType[any]) Set(arg string) error {
	tType := t.String()
	switch tType {
	case "float64":
		var temp interface{}
		temp, err := strconv.ParseFloat(arg, 64)
		if err != nil {
			return numError(err)
		}
		t.val = temp.(any)
	case "string":
		var temp interface{}
		temp = arg
		t.val = temp.(any)
	}
	t.set = true
	return nil
}
```
I was having trouble directly setting `t.val` because of type mismatch. So I came up with above workaround. 
Please let me know if this is the correct approach.

Reply:
it's on the right track. i think you should be able to do just this, for example:

```go
switch tType {
case "float64":
    temp, err := strconv.ParseFloat(...)
    if err != nil { ... }
    t.val = temp.(any)
```

Reply:
I tried that earlier but was given InvalidAssert.
`invalid operation: temp (variable of type float64) is not an interface`
Same error when I did:
`t.val = arg.(any)`

Reply:
ah then i guess you still need the declaration (but you should be able to declare as `any` rather than the older `interface{}`)

Reply:
On declaring any I still get
`cannot use strconv.ParseFloat(arg, 64) (value of type float64) as any/* type parameter */ value in assignment`

Reply:
https://go.dev/play/p/16iecA5dkc3

Reply:
Sorry but I could not figure out what is wrong. Below is a screenshot.
![image](https://github.com/vitessio/vitess/assets/21123621/de28a2ff-a4ba-43f9-8580-96493ab84e9c)

Reply:
i don't think from the code you've shown that you have the right definition for `Optional`. does it match what i suggested [here](https://github.com/vitessio/vitess/pull/14645#discussion_r1410796207)?

Reply:
Yeah I already rewrote it based on your suggestion.
![image](https://github.com/vitessio/vitess/assets/21123621/3084c0d0-2be1-4930-bea7-802b572b35f0)

Reply:
this approach seems to work: https://go.dev/play/p/RrQxZeRXD2a